File: Kaskus/GridViewPaging/GridViewPagingCs/Datas/RepositoryFactory.cs
namespace Dede.App.GridViewPaging.Datas
{
    using System;

    using Dede.App.GridViewPaging.Pages;

    public sealed class RepositoryFactory
    {
        private readonly IDataSourceRepository dataSourceRepository;

        public RepositoryFactory(
            IDataSourceRepository dataSourceRepository)
        {
            this.dataSourceRepository = dataSourceRepository;
        }

        public DataSourceResult GetData(
            Int32 pageSize, 
            Int32 currentPage)
        {
            return this.dataSourceRepository.GetData(pageSize, currentPage);
        }
    }
}

File: Kaskus/GridViewPaging/GridViewPagingVb/Datas/RepositoryFactory.vb
Imports System
Imports Dede.App.GridViewPaging.Pages

Namespace Datas

    Public NotInheritable Class RepositoryFactory
        Private ReadOnly _dataSourceRepository As IDataSourceRepository

        Public Sub New(dataSourceRepository As IDataSourceRepository)
            _dataSourceRepository = dataSourceRepository
        End Sub

        Public Function GetData(pageSize As Int32, currentPage As Int32) As DataSourceResult
            Return _dataSourceRepository.GetData(pageSize, currentPage)
        End Function
    End Class
End Namespace